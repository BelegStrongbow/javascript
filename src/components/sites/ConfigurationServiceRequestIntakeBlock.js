import PropTypes from "prop-types";
import React from "react";
import { defineMessages, FormattedMessage, injectIntl, intlShape } from "react-intl";
import { Paper, WhitePage } from "../PaperStyles";
import { LargeButton, makeButtonFullWidth } from "../Button";
import CollapsibleHeader from "../CollapsibleHeader";
import styled from "styled-components";

let messages = defineMessages( {
	configurationHeading: {
		id: "requestConfigurationIntake.configurationHeading",
		defaultMessage: "Configuration service intake form",
	},
	configurationAvailable: {
		id: "requestConfigurationIntake.configurationAvailable",
		defaultMessage: "You have purchased {configurationAvailable} available configuration " +
		"{configurationAvailable, plural, =0 {services. If you are interested in having your plugin configured by our experts, visit our store!}" +
		" one {service, which you can apply to one of your websites.} other {services, which you can apply to your websites.}}",
	},
	configurationHowTo: {
		id: "requestConfigurationIntake.configurationHowTo",
		defaultMessage: "Let us help you to configure the plugins for your website. Click on the button below to open the intake form.",
	},
	requestButton: {
		id: "requestConfigurationIntake.modalButton",
		defaultMessage: "Request configuration service for this site",
	},
} );

const ResponsiveButton = styled( makeButtonFullWidth( LargeButton ) )`
	white-space: nowrap;
	min-width: initial;
`;

/**
 * Returns the rendered ConfigurationServiceRequestBlock component.
 *
 * @param {Object} props The props to use.
 *
 * @returns {ReactElement} The rendered ConfigurationServiceRequestBlock component.
 */
class ConfigurationServiceRequestIntakeBlock extends React.Component {
	/**
	 * Sets the ConfigurationServiceRequestBlock component.
	 *
	 * @param {Object} props All of the props passed to this component.
	 *
	 * @returns {void}
	 */
	constructor( props ) {
		super( props );

		this.handleSubmit = this.handleSubmit.bind( this );
	}

	/**
	 * Handles submitting the form, passing the selectedOption to the callback.
	 *
	 * @returns {void}
	 */
	handleSubmit() {
		let siteId = this.props.siteId;
		this.props.openConfigurationServiceRequestModal( siteId );
	}

	render() {
		let configAvailable = this.props.amountAvailable;
		return (
			<Paper>
				<CollapsibleHeader
					title={ this.props.intl.formatMessage( messages.configurationHeading ) }
					isOpen={ true }
				>
					<WhitePage>
						<div>
							<p>
								<FormattedMessage
									id={ messages.configurationAvailable.id }
									defaultMessage={ messages.configurationAvailable.defaultMessage }
									values={ {
										configurationAvailable: configAvailable,
									} }
								/>
							</p>
							<p>
								<FormattedMessage
									id={ messages.configurationHowTo.id }
									defaultMessage={ messages.configurationHowTo.defaultMessage }
								/>
							</p>
						</div>
						<ResponsiveButton
							enabledStyle={ true }
							onClick={ this.handleSubmit }
						>
							{ this.props.intl.formatMessage( messages.requestButton ) }
						</ResponsiveButton>
					</WhitePage>
				</CollapsibleHeader>
			</Paper>
		);
	}
}

ConfigurationServiceRequestIntakeBlock.propTypes = {
	openConfigurationServiceRequestModal: PropTypes.func,
	amountAvailable: PropTypes.number.isRequired,
	intl: intlShape.isRequired,
	siteId: PropTypes.string.isRequired,
};

export default injectIntl( ConfigurationServiceRequestIntakeBlock );
