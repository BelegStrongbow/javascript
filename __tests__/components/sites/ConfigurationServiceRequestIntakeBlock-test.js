import React from 'react';
import ConfigurationServiceRequestIntakeBlock from '../../../src/components/sites/ConfigurationServiceRequestIntakeBlock';
import { createComponentWithIntl } from "../../../utils";

test('the configuration service request intake block component matches the snapshot', () => {
	const component = createComponentWithIntl(
		<ConfigurationServiceRequestIntakeBlock amountAvailable={ 1 } siteId="test" />
	);

	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test('the configuration service request intake block component calls the supplied function', () => {
	const mockupFunction = jest.fn();

	const component = createComponentWithIntl(
		<ConfigurationServiceRequestIntakeBlock amountAvailable={ 1 } siteId="test" openConfigurationServiceRequestModal={ mockupFunction } />
	);

	let tree = component.toJSON();

	// call onClick function on the component.
	const onClick = tree.children[0].children[1].children[1].props.onClick;
	onClick();

	expect(mockupFunction).toHaveBeenCalled();
});

test('the configuration service request status block with status completed component matches the snapshot', () => {
	const component = createComponentWithIntl(
		<ConfigurationServiceRequestIntakeBlock status="completed" />
	);

	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});
