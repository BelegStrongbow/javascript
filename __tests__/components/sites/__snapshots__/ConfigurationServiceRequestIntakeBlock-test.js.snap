// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the configuration service request intake block component matches the snapshot 1`] = `
.c0 {
  box-shadow: 0 2px 4px 0 rgba(0,0,0,0.2);
}

.c4 {
  background-color: #fff;
  padding: 16px 24px;
}

.c9 {
  height: 48px;
  margin: 0;
  padding: 0 16px;
  border: 0;
  border-bottom: 2px solid rgba(0,0,0,0.2);
  background-color: #64a60a;
  color: #fff;
  box-shadow: 0 2px 4px 0 rgba(0,0,0,0.2);
  border-radius: 4px;
  font: 400 14px/24px "Open Sans",sans-serif;
  text-transform: uppercase;
  cursor: pointer;
  vertical-align: top;
  -webkit-transition: background-color 150ms ease-out;
  transition: background-color 150ms ease-out;
}

.c9:hover,
.c9:focus {
  box-shadow: 0 4px 10px 0 rgba(0,0,0,0.2),inset 0 0 0 100px rgba( 0,0,0,0.1 );
}

.c9:active {
  -webkit-transform: translateY( 1px );
  -ms-transform: translateY( 1px );
  transform: translateY( 1px );
  box-shadow: none;
}

.c8 {
  text-shadow: 0px 0px 2px #000;
}

.c7 {
  min-width: 152px;
}

.c1 {
  margin-top: 24px;
  background-color: #fff;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #fff;
  padding: 24px 32px;
  border: none;
  text-align: left;
  font-weight: 300;
  cursor: pointer;
  color: #000;
}

.c2 svg {
  -webkit-flex: 0 0 40px;
  -ms-flex: 0 0 40px;
  flex: 0 0 40px;
  margin-right: -10px;
  margin-left: 10px;
  margin-top: -4px;
  margin-bottom: -4px;
  -webkit-align-self: flex-start;
  -ms-flex-item-align: start;
  align-self: flex-start;
}

.c3 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  font-size: 1.5em;
  line-height: 1.33333333;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c5 {
  white-space: nowrap;
  min-width: initial;
}

@media screen and ( max-width:800px ) {
  .c6 {
    width: 100%;
    background-image: none;
    padding: 12px 16px;
  }
}

@media screen and ( max-width:1355px ) {
  .c2 {
    padding: 16px 24px;
  }
}

@media screen and ( max-width:800px ) {
  .c2 {
    padding: 16px;
  }
}

@media screen and ( max-width:1355px ) {
  .c2 svg {
    margin-top: 4px;
    margin-bottom: 4px;
    margin-right: -2px;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <button
      aria-expanded={true}
      className="c2"
      onClick={[Function]}
    >
      <span
        className="c3"
      >
        Request configuration service
      </span>
      <svg
        aria-hidden="true"
        fill="#555"
        focusable="false"
        height={40}
        role="img"
        viewBox="0 0 24 24"
        width={40}
      >
        <path
          d="M7.41 15.41L12 10.83l4.59 4.58L18 14l-6-6-6 6z"
        />
      </svg>
    </button>
    <div
      className="c4"
    >
      <div>
        <p>
          <span>
            You have purchased 1 available configuration service, which you can apply to one of your websites.
          </span>
        </p>
        <p>
          <span>
            Let us help you configure the plugins for your website. Click on the button below to open the intake form.
          </span>
        </p>
      </div>
      <button
        className="c5 c6 c7 Button-kDSBcD c8 c9"
        onClick={[Function]}
        type="button"
      >
        Request configuration service for this site
      </button>
    </div>
  </div>
</div>
`;
